### PR TITLE
Add <Textfield /> Component

### LINKED ISSUE #32: Create <TextField /> Component
[Design System Link](https://www.notion.so/h4iuiuc/Hack4Impact-s-Design-System-Bridge-40aaae97c71f400084b4bbcc5cd8cdee#f529ec92de354584b6d0e0325c886342)

- TextField
- ~~TextField.Label~~

image

### Props
- `required`: `required`, `optional`
- `helperText`: str
- `autoComplete`
- `aria-label`
- `placeholder`
- `icon`?

more?

### PR DESCRIPTION
**Pull request by jackieo5023**
Resolves #32 and #33.  This also changes the card width to be consistent with the figma, and changes the cursor to pointer when hovering a button.

* Border on regular textfield is actually 'light' grey (on the figma) rather than 'lighter' as specified on the mock, so I made it 'light' in the code as well
* I only listed options for the `type` prop for stuff that doesn't look weird
* This expects a React Element to be passed into the `icon` prop, rather than just putting in an SVG as a child - since there can be an SVG as a result of setting the `error` prop
* The user can specify if they want a `textarea` by setting the `as` prop to "textarea", and optionally setting the number of lines they want with the `lines` prop
    * The red outlined text area shows the lines set to 2, while the disabled text area is the default

image
I have no idea why there seems to be weird spaces here ^ as a side note...
image

### CODE REVIEW

File: src/components/TextField/TextField.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+  let padding = '0px 9px 0px 9px';
+  if (as && as === 'textarea') {
+    padding = '12px 9px 12px 9px';
+  }
+
+  return (
+    <TextFieldContainer
+      disabled={disabled}
+      lines={lines || 3}
```
Comment (tko22) [inline src/components/TextField/TextField.js:119]: wait does the default text field include 3 lines? or does that only apply to textarea

Comment (jackieo5023) [reply src/components/TextField/TextField.js:119]: this just applies to textarea - so the default text area is 3 lines.  this is the code in the container that determines the height: ``` height: ${(props) => (props.type && props.type === 'textarea'
    ? `${props.lines * 20 + 30}px`
    : '46px')};```

Comment (tko22) [reply src/components/TextField/TextField.js:119]: oh right im stupid

**Comment by jackieo5023** [comment]
I added some basic documentation in terms of defining props, which will hopefully be helpful later down the road.  I can add some more in a separate PR!

**Pull request merged**